### PR TITLE
Adding a parameter for exclusion of trashed files in GoogleDriveHook

### PR DESCRIPTION
**Pull request by alexkruc**
GoogleDriveHook is currently returning all the files with the specified naming, regardless if they are in the "trash folder" or not.
Taking inspiration from the following PR: #24446, I saw that the [Google Drive API ](https://developers.google.com/drive/api/v3/reference/files/list?apix_params=%7B%22q%22%3A%22name%20%3D%20%27abcd_test%27%20and%20trashed%3Dfalse%22%7D) has a parameter that we can add to the query (`trashed=false/true`) that specifies whether we want the API to return trashed files or not (default is `True` -> show trashed files).

This PR is adding this query param to the `GoogleDriveHook`.

related: #24446

### CODE REVIEW

File: airflow/providers/google/suite/hooks/drive.py [hunk 1]
```diff
@@ -128,31 +128,45 @@ def get_media_request(self, file_id: str) -> HttpRequest:
         request = service.files().get_media(fileId=file_id)
         return request
 
-    def exists(self, folder_id: str, file_name: str, drive_id: Optional[str] = None):
+    def exists(
+        self, folder_id: str, file_name: str, include_trashed: bool = True, drive_id: Optional[str] = None
```
Comment (eladkal) [inline airflow/providers/google/suite/hooks/drive.py]: `include_trashed` is also optional param is it not?

Comment (alexkruc) [reply airflow/providers/google/suite/hooks/drive.py]: It is :) this is why we set it with a default value..
Setting it with `Optional[]` and with default value is redundant, as if we don't pass it, it's passed with it's default value..

Am I missing something?

Comment (alexkruc) [reply airflow/providers/google/suite/hooks/drive.py]: Just to be sure, I checked the official documentation of typing on `Optional[]`
https://docs.python.org/3/library/typing.html#typing.Optional
They also state the following
```
An optional argument with a default does not require the Optional qualifier ....
```

**Review by eladkal — APPROVED** [review]

File: airflow/providers/google/suite/hooks/drive.py (hunk 1 above)
Comment (uranusjr) [inline airflow/providers/google/suite/hooks/drive.py]: Can we change the ordering to put `include_trashed` at the back? Right now this (unnecessarily) breaks backward compatibility if the user passes `drive_id` as a positional argument.

Maybe it’s best to make the new argument keyword-only as well, so future additions don’t break compatibility easily:

```python
def exists(
    self,
    folder_id: str,
    file_name: str,
    drive_id: Optional[str] = None,
    *,
    include_trashed: bool = True,
) -> bool:
```

Comment (alexkruc) [reply airflow/providers/google/suite/hooks/drive.py]: Good point, I missed that one. Thanks for letting me know :)
Also, a good point about making it keyword-only.

I've implemented your suggestion :)

**Review by eladkal — APPROVED** [review]

**Pull request merged**